Issue98fix

hokie228 opened the pull request:
This fixes the encoding of a partial response to a multi-op request where one succeeds but another fails.  The current implementation returns an error with no response, but Kafka expects Zookeeper to still respond with two results, one with a type of opCheck and another with type opError.  

This patch allows it to response to each part of a multi-op request individually, using the first error in the list as the overall result.  This fixes the NPE in Kafka described in Issue #98

heyitsanthony commented:
Changes look OK; have you tried the cross-checked integration tests with `go test -tags docker,zkdocker`?

heyitsanthony commented on zketcd.go:
```diff
@@ -470,15 +470,23 @@ func (z *zkEtcd) Sync(xid Xid, op *SyncRequest) ZKResponse {
 			bs[i] = z.mkSetDataTxnOp(req)
+			mresp.Ops[i].Header.Type = opSetData
 		case *CheckVersionRequest:
+			mresp.Ops[i].Header.Type = opCheck
```
move this after mkCheckVersionPathTxnOp to be consistent with the other cases?

hokie228 replied:
Suggested change has been made. 

I have not been able to get the xchk,docker tests to run yet, they just seem to hang.  I can't get them to run for the master branch either.

Do you have any instructions on setting that up?

hokie228 commented:
I have also fixed the integration cross-check tests and implemented the Multi-op test.  There are still some failures in the integration tests unrelated to the Multi-op changes than I will try to address later, but all the unit and integration tests pass now.  I think everything is ready to go.

F21 commented:
Any chance this can be merged soon?

mickymiek commented:
@heyitsanthony ?

arizvisa commented:
@xiang90, can you look at merging this? It seems to have been silently dropped by the original maintainer...
